Code/Tools/ProjectManager/Source/ProjectExportWorker.cpp

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <ProjectExportWorker.h>
+#include <ProjectManagerDefs.h>
+#include <PythonBindingsInterface.h>
+#include <ProjectUtils.h>
+
+#include <QDir>
+#include <QFile>
+#include <QProcess>
+#include <QProcessEnvironment>
+#include <QTextStream>
+#include <QThread>
+
+namespace O3DE::ProjectManager
+{
+    ProjectExportWorker::ProjectExportWorker(const ProjectInfo& projectInfo)
+        : QObject()
+        , m_projectInfo(projectInfo)
+    {
+    }
+
+    void ProjectExportWorker::ExportProject()
+    {
+        auto result = ExportProjectForPlatform();
+
+        if (result.IsSuccess())
+        {
+            emit Done();
+        }
+        else
+        {
+            emit Done(result.GetError());
+        }
+    }
+
+    QString ProjectExportWorker::GetLogFilePath() const
+    {
+        QDir logFilePath(m_projectInfo.m_path);
+        // Make directories if they aren't on disk
+        if (!logFilePath.cd(ProjectBuildPathPostfix))
+        {
+            logFilePath.mkpath(ProjectBuildPathPostfix);
+            logFilePath.cd(ProjectBuildPathPostfix);
+
+        }
+        if (!logFilePath.cd(ProjectBuildPathCmakeFiles))
+        {
+            logFilePath.mkpath(ProjectBuildPathCmakeFiles);
+            logFilePath.cd(ProjectBuildPathCmakeFiles);
+        }
+        return logFilePath.filePath(ProjectExportErrorLogName);
+    }
+
+    void ProjectExportWorker::QStringToAZTracePrint([[maybe_unused]] const QString& error)
+    {
+        AZ_TracePrintf("Project Manager", error.toStdString().c_str());
+    }
+
+    //TODO: update this code to just run the desired worker process with the project path argument
+    AZ::Outcome<void, QString> ProjectExportWorker::ExportProjectForPlatform()
+    {
+        // Check if we are trying to cancel task
+        if (QThread::currentThread()->isInterruptionRequested())
+        {
+            QStringToAZTracePrint(ExportCancelled);
+            return AZ::Failure(ExportCancelled);
+        }
+
+        QFile logFile(GetLogFilePath());
+        if (!logFile.open(QIODevice::WriteOnly | QIODevice::Text | QIODevice::Truncate))
+        {
+            QString error = tr("Failed to open log file.");
```

Comment:
If this is a fixed string, why not make it constant like `ExportCancelled` is. Otherwise, I'd recommend using QString formatting to include the name of the log that failed to open.